### PR TITLE
Add configuration port validation

### PR DESCRIPTION
**Pull request by knesenko**
Signed-off-by: Kiril Nesenko <knesenko@vmware.com>

### CODE REVIEW

File: config/load.go [hunk 1]
```diff
@@ -18,6 +19,9 @@ func Load(configFile string) (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
+	if (config.HttpConfig.Port < 0) || (config.HttpConfig.Port > 65535) || (config.HttpsConfig.Port < 0) || (config.HttpsConfig.Port > 65535) {
```
Comment (asafka) [inline config/load.go]: This is a utility method - not the place to interpret / validate the config. Please move this logic to Server.initHttp() and Server.initHttps(). You will need to add 'error' to the return param list of initHttp().

Comment (knesenko) [inline config/load.go]: will fix

File: server/server.go [hunk 2]
```diff
@@ -165,18 +167,28 @@ func (server *Server) firstTimeInit(configuration *config.Config) error {
...
-func (server *Server) initHttp(configuration *config.Config) {
+func (server *Server) initHttp(configuration *config.Config) error {
 	server.useHttp = true
 	httpPort := configuration.HttpConfig.Port
+	if err := util.CheckPort(httpPort); err != nil {
+		return err
+	}
+
 	server.httpPort = DefaultHttpPort
 	if httpPort != 0 {
```
Comment (asafka) [inline server/server.go]: I would remove this check if util.CheckPort( ) requires the port to be > 0 (not >= 0) (which is probably what we want anyway).

File: server/server.go [hunk 3]
```diff
@@ -165,18 +167,28 @@ func (server *Server) firstTimeInit(configuration *config.Config) error {
...
 
 func (server *Server) initHttps(configuration *config.Config) error {
 	server.useHttps = true
 	httpsPort := configuration.HttpsConfig.Port
+	if err := util.CheckPort(httpsPort); err != nil {
+		return err
+	}
+
 	server.httpsPort = DefaultHttpsPort
 	if httpsPort != 0 {
```
Comment (asafka) [inline server/server.go]: Similar to above

File: util/util.go [hunk 4]
```diff
@@ -102,3 +102,11 @@ func GetChildSearchPattern(path string) string {
 		return path + "/?*"
 	}
 }
+
+func CheckPort(port int) error {
+	if (port < 0) || (port > 65535) {
```
Comment (asafka) [inline util/util.go]: See comments above - it's better to check port <= 0 here.

File: server/server.go (hunk 2 above)
Comment (knesenko) [inline server/server.go]: so what you suggest is not using default port at all and using the value from the config file always?

Comment (asafka) [inline server/server.go]: No...I guess we can keep 0 to denote "default", or (a bit more elegant) we can use the label "default" to return the default port. Not critical for now really.

**Pull request merged**